### PR TITLE
Add Tennix

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] ~~I am an upstream contributor to the project.~~ If not, I contacted upstream developers about submitting their software to Flathub.
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
Ping @thp. Can you put `tennix.appdata.xml` into https://repo.or.cz/tennix.git/tree/HEAD:/data?

**Comment by Mailaender** [comment]
bot, build org.icculus.tennix

**Comment by Mailaender** [comment]
bot, build org.icculus.tennix

**Comment by Mailaender** [comment]
bot, build org.icculus.tennix

**Comment by Mailaender** [comment]
Had to cut out the icon from `tennix.svg` as well to get a square one.

**Comment by hfiguiere** [comment]
> Had to cut out the icon from `tennix.svg` as well to get a square one.

you could patch it. :-)

File: org.icculus.tennix.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  - shared-modules/SDL/SDL_ttf-2.0.11.json
+
+  - name: tennix
+    buildsystem: simple
+    build-commands:
+      - ./configure --prefix ${FLATPAK_DEST}
+      - make install
+      - install -Dm0644 data/tennix.desktop ${FLATPAK_DEST}/share/applications/tennix.desktop
+      - install -Dm0644 tennix.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/tennix.svg
+      - install -Dm0644 tennix.appdata.xml ${FLATPAK_DEST}/share/metainfo/tennix.appdata.xml
```
Comment (hfiguiere) [inline org.icculus.tennix.yaml:39]: I will reiterate what I said previously in previous reviews: you should have it properly named, with the proper app-id.

`tennix.desktop` is NO LONGER a valid app-id.

File: tennix.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<application>
+  <id type="desktop">tennix.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
```
Comment (hfiguiere) [inline tennix.appdata.xml:4]: no project license?

File: tennix.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<application>
+  <id type="desktop">tennix.desktop</id>
```
Comment (hfiguiere) [inline tennix.appdata.xml:3]:
```suggestion
<component type="desktop-application">
  <id>org.icculus.tennix</id>
```

and you need to fix the closing tag.

File: org.icculus.tennix.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  - --socket=pulseaudio
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --device=dri
+
+modules:
+  - shared-modules/glu/glu-9.json
+
+  - name: sdl12-compat
```
Comment (hfiguiere) [inline org.icculus.tennix.yaml:20]: don't you miss some cleanup rules for `/include` and maybe `*.a` ?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.icculus.tennix.yaml (hunk 1 above)
Comment (Mailaender) [reply org.icculus.tennix.yaml:39]: This is a very old piece of software. It is way more practical to use the Flatpak rename scripts than sedding all the files and its mentionings manually.

Comment (barthalion) [reply org.icculus.tennix.yaml:39]: You literally need to replace `tennix` with ID or `$FLATPAK_ID` in these 3 lines.

Comment (hfiguiere) [reply org.icculus.tennix.yaml:39]: My comment is just for the appdata. `rename-icon` will rename in the .desktop, and `rename-desktop-file` will rename the launchable in the appdata, so there is a benefit for either.
However the appdata has no dependency so it should be written with the proper app-id and have the proper name.

**Comment by thp** [comment]
Upstream author here.

I'm happy to merge patches for the upstream repo to make packaging easier including e.g. providing a SVG icon without having to resort to patches :)

**Comment by Mailaender** [comment]
bot, build org.icculus.tennix

**Pull request closed without merging**